From: Robert Schwebel <r.schwebel@pengutronix.de>
Subject: [samba] make oplock support conditional

Linux oplock support is conditional on HAVE_KERNEL_OPLOCKS_LINUX, not
plain old LINUX.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---
 source/modules/vfs_default.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

# 20100428 rsc: This patch was already applied in the samba upstream, see
# http://websvn.samba.org/cgi-bin/viewcvs.cgi?view=rev&root=samba&rev=23962
# and d8d84cd7082a95e673a7efc285879dd412b78b4c in samba git.

diff --git a/source/modules/vfs_default.c b/source/modules/vfs_default.c
index 4cc6e88..1f8fb63 100644
--- a/source/modules/vfs_default.c
+++ b/source/modules/vfs_default.c
@@ -813,10 +813,11 @@ static int vfswrap_linux_setlease(vfs_handle_struct *handle, files_struct *fsp,
 
 	START_PROFILE(syscall_linux_setlease);
 
-#ifdef LINUX
+#ifdef HAVE_KERNEL_OPLOCKS_LINUX
 	/* first set the signal handler */
-	if(linux_set_lease_sighandler(fd) == -1)
+	if(linux_set_lease_sighandler(fd) == -1) {
 		return -1;
+	}
 
 	result = linux_setlease(fd, leasetype);
 #else
-- 
1.6.3.3

